### PR TITLE
Added typescript declaration files.

### PR DESCRIPTION
**Pull request by bmass02**
I wanted to use your library in a TS + Vue project, but it didn't have any TS declarations. Thought I'd share.

This will add the ```bindFirebaseRef``` and ```unbindFirebaseRef``` functions to the action's context typing.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import * as Vuex from 'vuex';
+import * as firebase from 'firebase';
+export declare const firebaseMutations: {
```
Comment (kaorun343) [inline src/index.d.ts]: You can write like this, I think.

```ts
export declare const firebaseMutations: Record<string, any>;
```

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    [key: string]: any;
+};
+export declare namespace VuexFire {
+    interface BindOptions {
+        cancelCallback?: Function;
+        readyCallback?: ((a: firebase.database.DataSnapshot, b?: string | undefined) => any);
+        errorCallback?: Function;
+        wait?: Boolean;
```
Comment (kaorun343) [inline src/index.d.ts]: `Boolean` -> `boolean`

**Comment by posva** [comment]
Wow, thanks a lot @bmass02 @kaorun343 
Do I need to add anything else to the build step or do I need to add some tests?

**Comment by kaorun343** [comment]
Test files are needed as well as Vue's projects, I think.

**Pull request merged**